src/geometry/MeshDrawMgr.cxx

```diff
@@ -84,16 +74,67 @@ MeshDrawMgr::~MeshDrawMgr()
     return;
 }
 
+void MeshDrawMgr::initVBO()
+{
+    vboArrayIndex = vboVTN.vboAlloc(vertexCount);
+    vboVTN.vertexData(vboArrayIndex, vertexCount, vertices);
+    vboVTN.textureData(vboArrayIndex, vertexCount, texcoords);
+    vboVTN.normalData(vboArrayIndex, vertexCount, normals);
+
+    for (int lod = 0; lod < lodCount; lod++)
+    {
+        const DrawLod& drawLod = drawLods[lod];
+        for (int set = 0; set < drawLod.count; set++)
+        {
+            const DrawSet& drawSet = drawLod.sets[set];
+            const int cmdCount = drawSet.count;
+            GLuint  indices[8192];
+            GLuint *indexP = indices;
+            int     fillP  = 0;
+            for (int i = 0; i < cmdCount; i++)
+            {
+                const DrawCmd& cmd = drawSet.cmds[i];
+                fillP += cmd.count;
+                if (fillP > 8192)
+                {
+                    std::cout << "MeshDrawMgr cmdSet " << fillP << " is too high" << std::endl;
+                    abort();
+                }
+                if (cmd.indexType == GL_UNSIGNED_SHORT)
+                {
+                    GLushort *shortP = (GLushort *)cmd.indices;
+                    for (int j = 0; j < cmd.count; j++)
+                        *indexP++ = (GLuint)*shortP++;
+                }
+                else if (cmd.indexType == GL_UNSIGNED_INT)
+                {
+                    memcpy(indices, cmd.indices, cmd.count * sizeof(GLuint));
```

Comment:
`memcpy` is evil.
```suggestion
                    auto src = static_cast<GLuint*>(cmd.indices);
                    std::copy(src, src + cmd.count, indices);
```
(assuming you have validated that cmd.count is less than the size of indices)

By the way, given the handling of the ushort case, should this really be
```suggestion
                    auto src = static_cast<GLuint*>(cmd.indices);
                    indexP = std::copy(src, src + cmd.count, indexP);
```
?